From: Simon McVittie <smcv@debian.org>
Date: Wed, 5 Aug 2020 11:00:01 +0100
Subject: Install session.conf.in to expected path for installed-tests

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://github.com/flatpak/xdg-desktop-portal/pull/521
---
 tests/Makefile.am.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/Makefile.am.inc b/tests/Makefile.am.inc
index e4ea3de..9aed3f1 100644
--- a/tests/Makefile.am.inc
+++ b/tests/Makefile.am.inc
@@ -109,7 +109,7 @@ dist_installed_test_dbs_DATA = tests/dbs/no_tables
 dist_installed_test_portals_DATA = tests/portals/test.portal
 endif
 
-dist_installed_test_data = \
+dist_installed_test_DATA = \
 	tests/session.conf.in \
 	$(NULL)
 
