From: Simon McVittie <smcv@debian.org>
Date: Fri, 25 Jan 2019 20:35:42 +0000
Subject: validate-icon: Add a #define for bwrap

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 src/Makefile.am.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am.inc b/src/Makefile.am.inc
index 8641710..6a6da59 100644
--- a/src/Makefile.am.inc
+++ b/src/Makefile.am.inc
@@ -217,4 +217,4 @@ xdg_desktop_portal_CPPFLAGS = \
 
 xdg_desktop_portal_validate_icon_SOURCES = src/validate-icon.c
 xdg_desktop_portal_validate_icon_LDADD = $(GDK_PIXBUF_LIBS)
-xdg_desktop_portal_validate_icon_CFLAGS = $(GDK_PIXBUF_CFLAGS)
+xdg_desktop_portal_validate_icon_CFLAGS = $(GDK_PIXBUF_CFLAGS) -DHELPER=\"$(BWRAP)\"
